### PR TITLE
Add module kallisto/index

### LINKED ISSUE #346: new module: KALLISTO/INDEX (for scRNAseq)
I think it would be good to have a module for [KALLISTO](https://bioconda.github.io/recipes/kallisto/README.html)

- [x] This module does not exist yet with the [`nf-core modules list`](https://github.com/nf-core/tools#list-modules) command
- [x] There is no [open pull request](https://github.com/nf-core/modules/pulls) for this module
- [x] There is no [open issue](https://github.com/nf-core/modules/issues) for this module
- [x] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

Example process:
https://github.com/nf-core/scrnaseq/blob/13b7b90fb452108b9423523402a140331761a672/main.nf#L354

### PR DESCRIPTION
**Pull request by ggabernet**
## PR checklist

Closes #346   

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `<SOFTWARE>.version.txt` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

**Review by FloWuenne — APPROVED** [review]
Ran local tests as well, all passing. @drpatelh might want the TODOs in the meta.yml removed?

Otherwise LGTM!

File: software/kallisto/index/main.nf [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    conda (params.enable_conda ? "bioconda::kallisto=0.46.2" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/kallisto:0.46.2--h4f7b962_1"
+    } else {
+        container "quay.io/biocontainers/kallisto:0.46.2--h4f7b962_1"
+    }
+
+    input:
+    path(fasta)
```
Comment (drpatelh) [inline software/kallisto/index/main.nf:22]:
```suggestion
    path fasta
```

File: software/kallisto/index/main.nf [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    output:
+    tuple path("*.idx"), emit: idx
+    path "*.version.txt"          , emit: version
```
Comment (drpatelh) [inline software/kallisto/index/main.nf:26]:
```suggestion
    tuple path("*.idx") , emit: idx
    path "*.version.txt", emit: version
```

Comment (drpatelh) [reply software/kallisto/index/main.nf:26]: Be nice if we can output a directory instead like with Salmon and all of the other aligners: https://github.com/nf-core/modules/blob/acb1a12a56f6eca1d9a30b86ca880d39af0ba5d6/software/salmon/index/main.nf#L26

**Review by drpatelh — CHANGES_REQUESTED** [review]
Minor changes @ggabernet ! Thanks!

File: software/kallisto/index/main.nf (hunk 2 above)
Comment (ggabernet) [reply software/kallisto/index/main.nf]: Ok sure, I'll do it like that then

**Comment by ggabernet** [comment]
I've added your changes @drpatelh , thanks for the review!

**Review by drpatelh — APPROVED** [review]
Thank you!

**Pull request merged**